libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py

```diff
@@ -531,3 +531,26 @@ def _start_chat(
             )
         else:
             return self.client.start_chat(message_history=history.history, **kwargs)
+
+    def get_num_tokens_from_messages(self, messages: List[BaseMessage]) -> int:
```

Comment:
it's probably better to move the `get_num_tokens` method from `VertexAI` to `_VertexAICommon`, and then everything should work just fine.

Reply:
Done

Reply:
we don't even need `get_num_tokens_from_messages`, we need to move an existing `get_num_tokens` method and then it should work:
https://github.com/langchain-ai/langchain/blob/f027696b5f068bacad96a9356ae196f5de240007/libs/core/langchain_core/language_models/base.py#L275

Reply:
@lkuligin This approach would result in multiple calls to the count_token endpoint i.e separately for each message. This approach in this PR collates all text in messages and calls the  count_token endpoint only once. 
```
https://github.com/langchain-ai/langchain/blob/f027696b5f068bacad96a9356ae196f5de240007/libs/core/langchain_core/language_models/base.py#L275
```

Reply:
fair enough, but let's still move the `get_num_tokens`, please. 
and I'm not sure that difference (1 large call vs n smaller calls) is significant.

@sasha-gitg any thoughts on this?

Reply:
additional thoughts: 
1. let's move `get_num_tokens` pls
2. you can always collapse message together:
either with
```
from vertexai.generative_models._generative_models import _PreviewGenerativeModel

get_buffer_string(
    messages, 
    human_prefix=_PreviewGenerativeModel._USER_ROLE,
    ai_prefix=_PreviewGenerativeModel._MODEL_ROLE)
```
or with 
```
self.client..count_tokens([m.content for m in messages])
```